python/tests/operations/abs.py

```diff
@@ -0,0 +1,218 @@
+                samples=block.samples,
+                components=block.components,
+                properties=block.properties,
+                values=np.abs(block.values),
+            )
+            sign_values = np.sign(block.values)
+            shape = ()
+            for c in block.components:
+                shape += (len(c),)
+            shape += (len(block.properties),)
+            for parameter, gradient in block.gradients():
+                diff_components = len(gradient.components) - len(block.components)
+                new_grad = gradient.data[:] * sign_values[
+                    gradient.samples["sample"]
+                ].reshape((-1,) + (1,) * diff_components + shape)
+                new_block.add_gradient(
+                    parameter,
+                    new_grad,
+                    gradient.samples,
+                    gradient.components,
+                )
+
+            abs_blocks.append(new_block)
+
+        tensor_result = TensorMap(A.keys, abs_blocks)
+        assert equistore.equal(tensor_result, tensor_abs)
+
+    def test_self_abs_tensor_complex(self):
+        block_1 = TensorBlock(
+            values=np.array([[1 + 2j, -2j], [5 - 3j, -5 + 5j]]),
+            samples=Labels(["samples"], np.array([[0], [2]], dtype=np.int32)),
+            components=[],
+            properties=Labels(["properties"], np.array([[0], [1]], dtype=np.int32)),
+        )
+        block_2 = TensorBlock(
+            values=np.array([[-1 + 1j, -2 + 6j], [4 - 3j, -4], [5j, 3]]),
+            samples=Labels(["samples"], np.array([[0], [2], [7]], dtype=np.int32)),
+            components=[],
+            properties=Labels(["properties"], np.array([[0], [1]], dtype=np.int32)),
+        )
+
+        block_res1 = TensorBlock(
+            values=np.array(
+                [
+                    [np.sqrt(1 + 2**2), 2],
+                    [np.sqrt(5**2 + 3**2), np.sqrt(5**2 + 5**2)],
+                ]
+            ),
+            samples=Labels(["samples"], np.array([[0], [2]], dtype=np.int32)),
+            components=[],
+            properties=Labels(["properties"], np.array([[0], [1]], dtype=np.int32)),
+        )
+        block_res2 = TensorBlock(
+            values=np.array(
+                [[np.sqrt(1 + 1), np.sqrt(2**2 + 6**2)], [5, 4], [5, 3]]
+            ),
+            samples=Labels(["samples"], np.array([[0], [2], [7]], dtype=np.int32)),
+            components=[],
+            properties=Labels(["properties"], np.array([[0], [1]], dtype=np.int32)),
+        )
+        keys = Labels(
+            names=["key_1", "key_2"], values=np.array([[0, 0], [1, 0]], dtype=np.int32)
+        )
+        A = TensorMap(keys, [block_1, block_2])
+        tensor_abs = equistore.abs(A)
+        tensor_result = TensorMap(keys, [block_res1, block_res2])
+        print("oo", tensor_result[0].values)
+        print("pp", tensor_abs[0].values)
+        assert equistore.equal(tensor_result, tensor_abs)
```

Comment:
This should be an allclose assert, since complex abs might have some [ULPS](https://en.wikipedia.org/wiki/Unit_in_the_last_place) differences with double sqrt.

Reply:
It is still weird that in both mine and @DivyaSuman14 's machine the test do not fail

Reply:
CI is using Intel mac, maybe both of you are using M1?

Reply:
I have intel

Reply:
But anyway I do not know I changed to allclose